Update test setup for changing [Feature:PetSet] to [Feature:StatefulSet]

janetkuo opened the pull request:
@fejta

Ref https://github.com/kubernetes/kubernetes/pull/37243

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/1151)
<!-- Reviewable:end -->

janetkuo commented on jobs/ci-kubernetes-e2e-gce-statefulset.sh:
```diff
@@ -31,8 +31,8 @@ export CLOUDSDK_CORE_PRINT_UNHANDLED_TRACEBACKS="1"
-export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:PetSet\]"
-export PROJECT="kubernetes-petset"
+export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:StatefulSet\]"
+export PROJECT="kubernetes-statefulset"
```
Is it okay to update PROJECT?

fejta replied:
Yes as long as this GCP project exists and is configured correctly... I would recommend leaving the projects alone for now.

janetkuo commented:
I didn't update these files since StatefulSet is beta, not alpha. 

```console
$ git grep [Pp]et[Ss]et
jobs/ci-kubernetes-e2e-gce-alpha-features-release-1.4.sh:export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:(ExternalTrafficLocalOnly|PetSet|DynamicKubeletConfig)\]|ScheduledJob"
jobs/ci-kubernetes-e2e-gce-alpha-features-release-1.5.sh:export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:(ExternalTrafficLocalOnly|PetSet|DynamicKubeletConfig)\]|ScheduledJob"
jobs/ci-kubernetes-e2e-gce-alpha-features.sh:export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:(ExternalTrafficLocalOnly|PetSet|DynamicKubeletConfig)\]|ScheduledJob"
jobs/ci-kubernetes-e2e-gci-gce-alpha-features-release-1.4.sh:export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:(ExternalTrafficLocalOnly|PetSet|DynamicKubeletConfig)\]|ScheduledJob"
jobs/ci-kubernetes-e2e-gci-gce-alpha-features-release-1.5.sh:export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:(ExternalTrafficLocalOnly|PetSet|DynamicKubeletConfig)\]|ScheduledJob"
jobs/ci-kubernetes-e2e-gci-gce-alpha-features.sh:export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:(ExternalTrafficLocalOnly|PetSet|DynamicKubeletConfig)\]|ScheduledJob"
jobs/ci-kubernetes-e2e-gci-gke-alpha-features-release-1.4.sh:export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:(ExternalTrafficLocalOnly|PetSet)\]|ScheduledJob"
jobs/ci-kubernetes-e2e-gci-gke-alpha-features.sh:export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:(ExternalTrafficLocalOnly|PetSet|DynamicKubeletConfig)\]|ScheduledJob"
jobs/ci-kubernetes-e2e-gke-alpha-features-release-1.4.sh:export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:(ExternalTrafficLocalOnly|PetSet)\]|ScheduledJob"
jobs/ci-kubernetes-e2e-gke-alpha-features-release-1.5.sh:export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:(ExternalTrafficLocalOnly|PetSet)\]|ScheduledJob"
jobs/ci-kubernetes-e2e-gke-alpha-features.sh:export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:(ExternalTrafficLocalOnly|PetSet|DynamicKubeletConfig)\]|ScheduledJob"
```

fejta commented on jobs/ci-kubernetes-e2e-gci-gce-statefulset.sh:
```diff
@@ -31,8 +31,8 @@ export CLOUDSDK_CORE_PRINT_UNHANDLED_TRACEBACKS="1"
-export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:PetSet\]"
-export PROJECT="kubernetes-gci-petset"
+export GINKGO_TEST_ARGS="--ginkgo.focus=\[Feature:StatefulSet\]"
+export PROJECT="kubernetes-gci-statefulset"
```
Same

janetkuo commented:
@fejta Once https://github.com/kubernetes/kubernetes/pull/37243 is merged, no 1.5 e2e tests will match Feature:PetSet

fejta commented:
Hmm... ok so if the test code in release-1.4 has petset but 1.5 and master have StatefulSet then I would leave the gce-alpha-features-release-1.4.sh with PetSet and update gce-alpha-features.sh and gce-alpha-features-release-1.5.sh to use StatefulSet.

fejta commented:
/lgtm
After making those changes.

fejta commented:
/lgtm

janetkuo commented:
@fejta StatefulSets are beta in 1.5. Should we just remove them from gce-alpha-features.sh and gce-alpha-features-release-1.5.sh?

krzyzacy commented:
I think if you remove it from alpha-features-release-1.5, you'll need to create a designated test, like gce-statefulset-release-1.5?

janetkuo commented:
In the 2nd commit I added StatefulSet to alpha feature tests, and replaced PetSet with StatefulSet in 1.5 alpha feature tests. 

But should we move StatefulSet to its own test, or **just remove all [Feature:PetSet]/[Feature:StatefulSet] tag** from it, since it's not alpha?

krzyzacy commented:
maybe @saad-ali can answer :-)

saad-ali commented:
> In the 2nd commit I added StatefulSet to alpha feature tests, and replaced PetSet with StatefulSet in 1.5 alpha feature tests.
> 
> But should we move StatefulSet to its own test, or just remove all [Feature:PetSet]/[Feature:StatefulSet] tag from it, since it's not alpha?

Why does it need a special test suite? Why not just move it to the [`slow` test suite](https://k8s-gubernator.appspot.com/builds/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-slow)?

As long as the test is running in some test suite (e.g. slow), I'll be happy. I don't want the tests to fall through the cracks and stop getting run in any test suite.

janetkuo commented:
@saad-ali Thanks for the reply! 

@krzyzacy @fejta I removed PetSet from 1.5 alpha features tests. I kept other alpha tests unchanged (have PetSet, but not StatefulSet). PTAL 

All StatefulSet feature tests will be running in below two groups:

```yaml
  - name: gce-statefulset
    test_group_name: kubernetes-e2e-gce-statefulset
  - name: gci-gce-statefulset
    test_group_name: kubernetes-e2e-gci-gce-statefulset
```

Other StatefulSet e2e tests will be running in either "slow" or "serial", see changes in kubernetes/kubernetes#37243

krzyzacy commented:
still lgtm, feel free to merge once [#37243](https://github.com/kubernetes/kubernetes/pull/37243) is merged

fejta commented:
Sen/Janet, please add me back if I am needed.

janetkuo commented:
#37243 is merged, my can merge this one now. 

I don't have the permission to merge this. @krzyzacy would you merge this?

fejta commented:
Merged
